src/runner/pythonrunner.cpp

```diff
@@ -1445,11 +1458,53 @@ QList<QObject*> PythonRunner::instantiate(const QString &pluginName)
     return allInterfaceList;
   }
   catch (const bpy::error_already_set&) {
-    MOBase::log::error("Failed to import plugin from {}.", pluginName);
+    MOBase::log::error("Failed to import plugin from {}.", identifier);
     throw pyexcept::PythonError();
   }
 }
 
+void PythonRunner::unload(const QString& identifier)
+{
+  auto it = m_PythonObjects.find(identifier);
+  if (it != m_PythonObjects.end()) {
+
+    GILock lock;
+
+    // We need to remove modules from the plugin from sys.module. At this point,
```

Comment:
I'm not sure I'm following what needs to be done here

Reply:
Would this be correct?
"We need to unload from sys.module all the modules loaded from the plugin folder."

Reply:
`sys.modules` is really just a list, there is no real way to "unload" modules in Python unfortunately. Removing it from `sys.modules` make it so the next time it's imported, the module will be reloaded.

In the context of MO2, what I am doing is removing all the modules that are in `sys.modules` whose path is under the given plugin, e.g. for `basic_games`, it will remove:

- `basic_games` itself
- `basic_games.games`, because it's under `basic_games`
- `basic_games.basic_features`, for the same reason.

Another example with 3rd party libs, for `installer_wizard` it will remove:

- `installer_wizard`,
- `installer_wizard.ui`,
- `wizard` because it's loaded from `installer_wizard/lib`
- `antlr4` because it's loaded from `installer_wizard/lib`
- etc.

The idea is to be able to load the plugin again after unloading and to have correct changes taken into account. Honestly, I don't think there is a way to do that 100% clean with Boost.Python, so I'm doing the best I can since this will only be used for plugin development.